### PR TITLE
Adds a traitor spawn minor antagonist role, allows traitors to spawn on some gamemodes.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Needs testmerge, I haven't actually tested this since its hard to test with only 1 person but it should probably be fine.

## About The Pull Request

Adds in a minor antagonist datum that will have a chance to spawn traitors. Has a 7% chance per person with a maximum of 5 traitors.
Chances of appearing at all (If this prob fails then 0 will appear):
 - 60% Heretics (Heretics are relatively weak and some traitors here and there to spice things up and make heretics not the only things to validhunt would be nice)
 - 15% Devil (Devil is shit with no one that needs powers, although this is not in rotation)

## Why It's Good For The Game

Adds a low chance for a few game modes to be a little spicier, and shakes up the sec meta a little.

## Changelog
:cl:
add: Traitors now have a low chance of spawning on heretics (60%), devil (15%).
code: Minor antagonist roles can now be configured to optionally use antagonist rep
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
This is basically dynamic but for some reason forced on people

**Comment by heepox** [comment]
> This is basically dynamic but for some reason forced on people

dynamic would be if it spawned round ending antags and such... its just vanilla traitors so its fine.

**Comment by eeSPee** [comment]
Are traitors allowed to voluntarily seek out/join cult?

**Comment by utilbird** [comment]
Heretic with syndi revolver 😏

**Comment by Ratvarr** [comment]
tator with desword in da cult
this will be very funne

**Comment by Archanial** [comment]
I don't know why do people upvote this. This is straight up bad idea. This is why what dynamic was intended to be. I would be super mad if I roll traitor and then round ended up heretic death or clock cult would make my escape impossible.

**Comment by PowerfulBacon** [comment]
> I don't know why do people upvote this. This is straight up bad idea. This is why what dynamic was intended to be. I would be super mad if I roll traitor and then round ended up heretic death or clock cult would make my escape impossible.

Round ending on station counts as escape

**Comment by eeSPee** [comment]
IMO, leave it at constant 2% on all game modes.

**Comment by utilbird** [comment]
> Round ending on station counts as escape

You get 10 minutes before the shuttle docks and another 2 before it leaves to quickly do your objectives, or retrieve hidden items. Maybe it could force call shuttle instead? Or do nothing if there are other antags, similar to wiz survivalists (as bad as they are)

**Comment by Isy232** [comment]
> IMO, leave it at constant 2% on all game modes.

Not wizard, I think its important for wizard rounds to end when the wizard/slaughter demons die since they likely killed most of the crew if they murderboned hard.

File: code/game/gamemodes/clock_cult/clockcult.dm [hunk 1]
```diff
@@ -41,6 +41,8 @@ GLOBAL_VAR(clockcult_eminence)
 	<span class='danger'>Servants</span>: Convert more servants and defend the Ark of the Clockwork Justicar!\n\
 	<span class='notice'>Crew</span>: Prepare yourselfs and destroy the Ark of the Clockwork Justicar."
 
+	allowed_special = list(/datum/special_role/traitor)
```
Comment (PowerfulBacon) [inline code/game/gamemodes/clock_cult/clockcult.dm]:
```suggestion
```
no

File: code/game/gamemodes/cult/cult.dm [hunk 2]
```diff
@@ -52,6 +52,8 @@
 
 	title_icon = "cult"
 
+	allowed_special = list(/datum/special_role/traitor)
```
Comment (PowerfulBacon) [inline code/game/gamemodes/cult/cult.dm]:
```suggestion
```

**Comment by PowerfulBacon** [comment]
Removed from cult and clockcult because hijack doesn't work on these gamemodes

**Comment by heepox** [comment]
could just make a ghost contractor midround antag too for even more spice, but this is a nice idea.

**Comment by ike709** [comment]
Update the CL.

**Comment by PowerfulBacon** [comment]
Done

**Comment by ike709** [comment]
Didn't seem to horrifically break anything.

**Pull request merged**